analyses/cluster-cell-calling/01-cluster-cell-calling.Rmd

```diff
@@ -0,0 +1,305 @@
+  data_file <- file.path(data_dir, glue::glue("seurat_obj_integrated_{integration_method}.rds"))
+  
+  } else if (data_dir_module == "upstream-analysis") {
+  data_dir <- file.path(root_dir, "analyses", data_dir_module, "results", glue::glue('03_Filter_object{suffix}'))
+  data_file <- file.path(data_dir, glue::glue("seurat_obj_merged_filtered.rds"))
+  } else {}
+
+
+# Create results_dir
+module_results_dir <- file.path(analysis_dir, "results")
+if (!dir.exists(module_results_dir)) {
+  dir.create(module_results_dir)}
+
+results_dir <- file.path(module_results_dir, glue::glue("01_cluster_cell_calling_{resolution}"))
+if (!dir.exists(results_dir)) {
+  dir.create(results_dir)}
+
+# Create plots directory
+module_plots_dir <- file.path(analysis_dir, "plots") 
+if (!dir.exists(module_plots_dir)) {
+  dir.create(module_plots_dir)}
+
+plots_dir <- file.path(module_plots_dir, glue::glue("01_cluster_cell_calling_{resolution}")) 
+if (!dir.exists(plots_dir)) {
+  dir.create(plots_dir)}
+
+source(paste0(root_dir, "/figures/scripts/theme_plot.R"))
+source(paste0(analysis_dir, "/util/function-cluster-cell-calling.R"))
+```
+
+```{r echo=FALSE, warning=FALSE}
+opts_chunk$set(fig.align='center',
+               external=TRUE,
+               echo=FALSE,
+               warning=FALSE,
+               fig.pos='H')
+a4width <- 8.3
+a4height <- 11.7
+```
+
+# Read seurat object
+First, we will use the seurat object as generated from the pipeline in the `r data_dir_module` module. 
+
+```{r read-object, echo=TRUE}
+seurat_obj <- readRDS(data_file)
+```
+
+# Cluster the cells
+
+Seurat applies a graph-based clustering approach, building upon initial strategies in [Macosko et al., 2015](https://www.cell.com/fulltext/S0092-8674(15)00549-8). Importantly, the distance metric which drives the clustering analysis (based on previously identified PCs) remains the same. However, our approach to partitioning the cellular distance matrix into clusters has dramatically improved. Our approach was heavily inspired by recent manuscripts which applied graph-based clustering approaches to scRNA-seq data [SNN-Cliq, Xu and Su, Bioinformatics, 2015](https://academic.oup.com/bioinformatics/article/31/12/1974/214505) and CyTOF data [PhenoGraph, Levine et al., Cell, 2015](https://pubmed.ncbi.nlm.nih.gov/26095251/). Briefly, these methods embed cells in a graph structure - for example a K-nearest neighbor (KNN) graph, with edges drawn between cells with similar feature expression patterns, and then attempt to partition this graph into highly interconnected ‘quasi-cliques’ or ‘communities’.
+
+As in PhenoGraph, we first construct a KNN graph based on the euclidean distance in PCA space, and refine the edge weights between any two cells based on the shared overlap in their local neighborhoods (Jaccard similarity). This step is performed using the FindNeighbors() function, and takes as input the previously defined dimensionality of the dataset (first 10 PCs).
```

Comment:
Should this section be updated to reflect that scATAC-seq methods use the `lsi` approach for reduction before clustering instead of the `pca` approach used for scRNA-seq?